src/main/java/graphql/schema/DataFetchingEnvironment.java

```diff
@@ -55,6 +56,12 @@
      */
     <T> T getArgument(String name);
 
+    Map<String, Map<String, Object>> getDirectiveArguments();
+
+    Map<String, Object> getDirectiveArguments(String directiveName);
+
+    <T> Optional<T> getDirectiveArgument(String directiveName, String argumentName);
```

Comment:
As much as Optional is a better pattern, its mixing styles on this class.  We already use null via getArgument() (because it predates Java 8) so I think we should stay consistent to the other methods in this interface

Reply:
The difference between this and `getArgument(argumentName)` is that in this case even the `directiveName` might be wrong, before we even check the `argumentName`. But I get your point., it does look weird.